src/main/java/com/google/sps/Matrix.java

```diff
@@ -0,0 +1,172 @@
+        neighbors += this.hasAliveNeighbor(row-1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row-1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column-1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column-1,tempMatrix) ? 1 : 0;
+
+        return neighbors;
+    }
+
+    private boolean hasAliveNeighbor(int row, int column, byte[][] tempMatrix) {
+        int rows = tempMatrix.length,
+            columns = tempMatrix[0].length;
+
+        boolean boundaryCheck = row >= 0 && row < rows && column >= 0 && column < columns;
+        return boundaryCheck && tempMatrix[row][column] != Cell.DEAD.getCode(); 
+    }
+
+
+    private void stringToMatrix(String stringMatrix) {
+
+        int[] commaIndexes = findCommaIndexes(stringMatrix);
+
+        int rows = Integer.parseInt(stringMatrix.substring(0, commaIndexes[0])),
+            columns = Integer.parseInt(stringMatrix.substring(commaIndexes[0]+1, commaIndexes[1]));
+
+        int currentCell = commaIndexes[1]+1;
+
+        this.matrix = new byte[rows][columns];
+
+        for (int row = 0; row < rows; row++) {
+            for (int column = 0; column < columns; column++) {
+                this.matrix[row][column] = (byte) stringMatrix.charAt(currentCell++);
+            }
+        }
+
+        this.printMatrix(this.matrix);
+
+    }
+
+    private int[] findCommaIndexes(String stringMatrix) {
+        int[] commaIndexes = new int[2];
+
+        boolean firstCommaFound = false;
+
+        for (int i=0; i<stringMatrix.length(); i++) {
+            if (stringMatrix.charAt(i) == ',') {
+                if (!firstCommaFound) {
+                    commaIndexes[0] = i;
+                    firstCommaFound = true;
+                }
+                else {
+                    commaIndexes[1] = i;
+                    break;
+                }
+            }
+        }
+        return commaIndexes;
+
+    }
+
+    private void printMatrix(byte[][] matrix) { 
+        int rows = matrix.length,
+            columns = matrix[0].length;
+
+        for (int row = 0; row < rows; row++) {
+            for (int column = 0; column < columns; column++) {
+                System.out.print((char) matrix[row][column] + " ");
+            }
+            System.out.println();
+        }
+    }
+
+    private byte[][] getMatrixCopy(byte[][] matrix) {
+        int rows = matrix.length,
+        columns = matrix[0].length;
+
+        byte[][] matrixCopy = new byte[rows][columns];
+
+        for (int row = 0; row < rows; row++) {
+            for (int column = 0; column < columns; column++) {
+                matrixCopy[row][column] = matrix[row][column];
+            }
```

Comment:
You might be able to simplify this a bit by using [Arrays.copyOf](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#copyOf(byte[],%20int)) (or some of the other built in array copy tools)

Reply:
Oh, I did not try with Arrays.copyOf, I use other methods like clone(), but the matrix was clone by reference, but I will try to use copyOf() :)